sound/soc/sof/ipc4-pcm.c

```diff
@@ -191,6 +191,60 @@ sof_ipc4_update_pipeline_state(struct snd_sof_dev *sdev, int state, int cmd,
  * prepare ioctl before the START trigger.
  */
 
+static int sof_ipc4_chain_dma_send_msg(struct snd_sof_dev *sdev,
+				       struct snd_sof_pcm_stream_pipeline_list *pipeline_list,
+				       bool allocate, bool enable)
+{
+	struct sof_ipc4_msg msg = {{ 0 }};
+	int i;
+
+	msg.primary = SOF_IPC4_MSG_TYPE_SET(SOF_IPC4_GLB_CHAIN_DMA);
+	msg.primary |= SOF_IPC4_MSG_DIR(SOF_IPC4_MSG_REQUEST);
+	msg.primary |= SOF_IPC4_MSG_TARGET(SOF_IPC4_FW_GEN_MSG);
+
+	/*
+	 * Copy the Link DMA ID, host DMA ID, fifo_size and the SCS bit settings from the pipelines.
+	 */
+	for (i = 0; i < pipeline_list->count; i++) {
+		struct snd_sof_pipeline *spipe = pipeline_list->pipelines[i];
```

Comment:
more than 1 pipeline ? msg.primary and msg.extension will be set multiple times

Reply:
@RanderWang this is fully intentional. primary has teh host DMA ID and link DMA ID and they come from different pipelines. The extension contains the fifo_size and only comes from the FE pipeline. The BE pipeline never sets it. So it will always be 0. @jsarha maybe capture it with a comment?